inspirehep/records/indexer/base.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+import logging
+
+from elasticsearch.helpers import bulk
+from flask import current_app
+from inspire_schemas.errors import SchemaNotFound, SchemaKeyNotFound
+from invenio_indexer.api import RecordIndexer
+from invenio_indexer.signals import before_record_index
+from jsonschema.exceptions import SchemaError, ValidationError
+from sqlalchemy.orm.exc import NoResultFound
+
+from invenio_search import current_search_client as es
+
+logger = logging.getLogger(__name__)
+
+
+class InspireRecordIndexer(RecordIndexer):
+    """Extend Invenio indexer to properly index Inspire records"""
+
+    def _prepare_record(self, record, index, doc_type):
+        data = record.dumps_for_es()
+
+        before_record_index.send(
+            current_app._get_current_object(),
+            json=data,
+            record=record,
+            index=index,
+            doc_type=doc_type,
+        )
+        return data
+
+    def _process_bulk_record_for_index(
+        self, record, version_type="external_gte", index=None, doc_type=None
+    ):
+        """Process basic data required for indexing record during bulk indexing
+
+        Args:
+            record(InspireRecord): Proper inspire record record object
+            version_type(str): Proper ES versioning type:
+                * internal
+                * external
+                * external_gt
+                * external_gte
```

Comment:
missing `args` (there should be an automated check for this). We should move to something a bit more automatic. Maybe https://github.com/agronholm/sphinx-autodoc-typehints ?

Reply:
:+1: